### DISCUSSION TITLE
WiX Bundle updates/patches are no longer added to the new Windows 11 "Windows Update - Update History"

### DISCUSSION THREAD

**Discussion by WalkerNA** [wixtoolset/issues#8600] [Questions]
Is there a bug in WiX which prevents a patched Bundle from appearing within the new Windows 11 `Windows Update >> Update History` page?

We have a product which will patch a WiX Bundle via: `<RelatedBundle Id="<Upgrade_Code>" Action="patch" />`.  When we install this patch on Windows 10, we can navigate to `Control Panel > Programs > Programs and Features > Installed Updates` and will find the entry declaring that the Patch/Update has been installed and can be uninstalled.  For example:

![Windows10_UpdateHistory](https://github.com/wixtoolset/issues/assets/35239937/d442365d-2bed-4582-9f63-65a56bd1760d)

However, when we install the same Bundle + Patch on a Windows 11 (23H2) system, there is no mention of the installed update via the new `Windows Update > Installed Updates` UI:

![Windows11_UpdateHistory](https://github.com/wixtoolset/issues/assets/35239937/3425036f-2796-4970-96c3-94b6f595accd)

This behavior holds true for Bundle upgrades with WiX 3.10, 3.14.1.5, and 5.0.0.

Is this defective behavior in WiX where it does not properly register the update with the new Windows 11 'Update History'?  Are there different settings that are required to enable this functionality?  Has Windows disabled this functionality for 3rd party updates starting with Windows 11?

Any assistance would be greatly appreciated.

**Comment by bevanweiss** [comment]
This looks like a decision from Microsoft.
The first view shows 'Programs and Features' > 'Installed Updates', suggesting it's Installed Updates for all currently installed 'Programs' and 'Features' (which would include 3rd party 'Programs').
The second view clearly shows 'Windows Updates' > 'Uninstall Updates', suggesting only 'Windows Updates' will be shown here.  Which would NOT include 3rd party 'Programs'.

I'm unsure which criteria Microsoft are using to do their filtering to only show their updates, I note that it includes not just Windows Updates, but also SQL Server updates.  
But it would definitely not be recommended to modify your installers to try to appear as though they are Windows Updates.

**Reply by WalkerNA** [reply]
Thank you for your prompt reply.  I will also note that additional confusion comes because the two wildly different dialogs appear after clicking the same `View installed updates` button from within `Control Panel > Programs` which would still suggest that the 'to be displayed' information should still be similar:

![ViewInstalledUpdates](https://github.com/wixtoolset/issues/assets/35239937/ff484cf0-145b-49d3-93f1-7c33f35d264c)

However, in this instance, I think the issue might actually be on the WiX side.  In addition to your example of SQL Server updates being present, I have also verified that basic WiX 5 MSP updates are present in the new UI:

![Windows11_UpdateHistory_MSP](https://github.com/wixtoolset/issues/assets/35239937/2fb7158e-ad1d-4701-aa51-c8454a846809)

Since SQL and basic MSP updates are not official Windows Updates, this certainly proves that the new Windows 11 UI is not intended for _just_ official Windows Updates.  

To me, the fact that WiX MSP updates still show up in this new UI while WiX Bundle updates no longer appear suggests that WiX Bundle packages may need code changes to support the new UI (a new WiX bug introduced by a Windows OS change).

I haven't found any documentation yet which explains the update criteria for showing up on this UI (registry entries, etc.).  It was very convenient having the the singular, Windows supported located a customer could go to in order to verify that not only a Bundle update was already installed, but which update was installed (Update 1? 7? [X]?).  I will keep investigating for now before submitting a defect.

**Reply by yuvnith** [reply]
Hey, @WalkerNA  In Windows 11, I am also not able to view the patches in the update section. Did you manage to resolve this issue with WiX versions v3?

@robmen  any idea?

**Reply by robmen** [reply]
@yuvnith, none of my customers have reported any interest in this change in Windows so it is not high on my list to investigate.

**Reply by rowbot38** [reply]
Hi @robmen, what would it take to move up your list? We (and some of our customers) have noticed the "change". 

I started to delve into registry areas but couldn't find any keys/values that would indicate filtering so presumably it's purely and deliberately programmatic on Microsoft's part. 

I raised a technical support case with MS some time ago on this but they fobbed me off with "it might be addressed in a future update".
